build-bench-env.sh

```diff
@@ -192,6 +198,29 @@ function write_version {  # name, git-tag, repo
   echo "$1: $2, $commit, $3" > "$devdir/version_$1.txt"
 }
 
+function partial_checkout {  # name, git-tag, directory, git repo, directory to download
+  phase "build $1: version $2"
+  pushd $devdir
+  if test "$rebuild" = "1"; then
+    rm -rf "$3"
+  fi
+  if test -d "$3"; then
+    echo "$devdir/$3 already exists; no need to git clone"
+    cd "$3"
+  else
+    mkdir "$3"
+    cd "$3"
+    git init
+    git remote add origin $4
+    git config extensions.partialClone origin
+    git fetch --depth=1 --filter=blob:none origin $2
+    git sparse-checkout set $5
+  fi
+  git checkout $2
+  git pull origin $2
```

Comment:
After running once some changes have come into LLVM this seems to pull down all the subsequent changes.  I wonder if we should replace the `git pull` with more magic? 

```suggestion
    git sparse-checkout set $5
  fi
  git fetch --depth=1 --filter=blob:none origin $2
  git checkout $2
  git reset origin/$2 --hard
```